labs/src/main/scala/org/scalalabs/basic/lab03/OptionExercise.scala

```diff
@@ -25,9 +25,14 @@ object OptionExercise01 {
    * - does not exist: 					"not existing"
    */
   def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
+          if (room > rooms.size) "not existing" 
+          else{
+            val workingRoom = rooms.get(room)
+            if (workingRoom.head.isEmpty) "empty"
+            else if (workingRoom.head.get=="locked") "not available"
+            else workingRoom.head.get
+          }  
   }
```

Comment:
we got the same solution! i think this is a cool problem b/c it really displays the return style of scala which i find to be pretty cool/novel